sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -199,44 +202,71 @@ public override void UpdateRetryPolicy(EventHubRetryPolicy newRetryPolicy)
 
-                var token = await AquireAccessTokenAsync(cancellationToken).ConfigureAwait(false);
-                using AmqpMessage request = MessageConverter.CreateEventHubPropertiesRequest(EventHubName, token);
-                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                        var token = await AquireAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                        using AmqpMessage request = MessageConverter.CreateEventHubPropertiesRequest(EventHubName, token);
+                        cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-                var stopWatch = Stopwatch.StartNew();
-                RequestResponseAmqpLink link = await ManagementLink.GetOrCreateAsync(_tryTimeout).ConfigureAwait(false);
+                        RequestResponseAmqpLink link = await ManagementLink.GetOrCreateAsync(UseMinimum(ConnectionScope.SessionTimeout, _tryTimeout.CalculateRemaining(stopWatch.Elapsed))).ConfigureAwait(false);
+                        cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-                // Send the request and wait for the response.
+                        // Send the request and wait for the response.
 
-                stopWatch.Stop();
-                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                        using AmqpMessage response = await link.RequestAsync(request, _tryTimeout.CalculateRemaining(stopWatch.Elapsed)).ConfigureAwait(false);
+                        cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                        stopWatch.Stop();
 
-                using AmqpMessage response = await link.RequestAsync(request, _tryTimeout.CalculateRemaining(stopWatch.Elapsed)).ConfigureAwait(false);
-                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                        // Process the response.
 
-                // Process the response.
+                        AmqpError.ThrowIfErrorResponse(response, EventHubName);
+                        return MessageConverter.CreateEventHubPropertiesFromResponse(response);
+                    }
+                    catch (Exception ex)
+                    {
+                        EventHubsEventSource.Log.GetPropertiesError(EventHubName, ex.Message);
+
+                        // Determine if there should be a retry for the next attempt; if so enforce the delay but do not quit the loop.
+                        // Otherwise, mark the exception as active and break out of the loop.
+
+                        ++failedAttemptCount;
+                        retryDelay = _retryPolicy.CalculateRetryDelay(ex, failedAttemptCount);
+
+                        if ((retryDelay.HasValue) && (!ConnectionScope.IsDisposed))
+                        {
+                            await Task.Delay(retryDelay.Value).ConfigureAwait(false);
```

Comment:
Maybe we should pass the cancellation token to the `Task.Delay` method.

Reply:
Yes.  Yes, we should.  Great catch.

Reply:
Adding across Event Hub client and consumers using a retry loop.